diff --git a/src/Unleashed/Helpers/SniffLocalCache.php b/src/Unleashed/Helpers/SniffLocalCache.php
index ad0792f..a8eaab3 100644
--- a/src/Unleashed/Helpers/SniffLocalCache.php
+++ b/src/Unleashed/Helpers/SniffLocalCache.php
@@ -14,7 +14,7 @@ use PHP_CodeSniffer\Files\File;
 final class SniffLocalCache
 {
     /** @var array<int, array<string, mixed>> */
-    private static $cache = [];
+    private static array $cache = [];
 
     /**
      * @return mixed
diff --git a/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php b/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php
index 71be9a5..f9492c7 100644
--- a/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php
+++ b/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php
@@ -11,11 +11,10 @@ use Unleashed\Helpers\UseStatements;
 
 final class FullyQualifiedGlobalFunctionsSniff implements Sniff
 {
-    /** @var bool */
-    public $onlyOptimizedFunctions = false;
+    public bool $onlyOptimizedFunctions = false;
 
     /** @var array<string, bool> */
-    private $optimizedFunctions = [
+    private array $optimizedFunctions = [
         // @see https://github.com/php/php-src/blob/PHP-7.4/Zend/zend_compile.c "zend_try_compile_special_func"
         'array_key_exists'     => true,
         'array_slice'          => true,
diff --git a/tests/expected_report.txt b/tests/expected_report.txt
index 0ab45ec..fc414f1 100644
--- a/tests/expected_report.txt
+++ b/tests/expected_report.txt
@@ -14,7 +14,7 @@ tests/input/ControlStructures.php                     17      2
 tests/input/doc-comment-spacing.php                   12      0
 tests/input/duplicate-assignment-variable.php         1       0
 tests/input/EarlyReturn.php                           6       0
-tests/input/example-class.php                         38      0
+tests/input/example-class.php                         41      0
 tests/input/forbidden-comments.php                    14      0
 tests/input/forbidden-functions.php                   5       0
 tests/input/fully-qualified-and-fallbacks.php         1       0
@@ -38,16 +38,16 @@ tests/input/strict-functions.php                      4       0
 tests/input/test-case.php                             7       0
 tests/input/trailing_comma_on_array.php               1       0
 tests/input/traits-uses.php                           11      0
-tests/input/type-hints.php                            5       0
+tests/input/type-hints.php                            6       0
 tests/input/UnusedVariables.php                       2       0
 tests/input/use-function.php                          2       0
 tests/input/use-ordering.php                          9       0
 tests/input/useless-semicolon.php                     2       0
 tests/input/UselessConditions.php                     23      0
 ----------------------------------------------------------------------
-A TOTAL OF 362 ERRORS AND 2 WARNINGS WERE FOUND IN 41 FILES
+A TOTAL OF 366 ERRORS AND 2 WARNINGS WERE FOUND IN 41 FILES
 ----------------------------------------------------------------------
-PHPCBF CAN FIX 303 OF THESE SNIFF VIOLATIONS AUTOMATICALLY
+PHPCBF CAN FIX 307 OF THESE SNIFF VIOLATIONS AUTOMATICALLY
 ----------------------------------------------------------------------
 
 
diff --git a/tests/fixed/example-class.php b/tests/fixed/example-class.php
index 0f98ce6..050d89c 100644
--- a/tests/fixed/example-class.php
+++ b/tests/fixed/example-class.php
@@ -18,14 +18,12 @@ class Example implements \IteratorAggregate
 {
     private const VERSION = \PHP_VERSION - (PHP_MINOR_VERSION * 100) - PHP_PATCH_VERSION;
 
-    /** @var int|null */
-    private $foo;
+    private ?int $foo = null;
 
     /** @var string[] */
-    private $bar;
+    private array $bar;
 
-    /** @var bool */
-    private $baz;
+    private bool $baz;
 
     /** @var ControlStructureSniff|int|string|null */
     private $baxBax;
diff --git a/tests/fixed/type-hints.php b/tests/fixed/type-hints.php
index 0e952fc..9824fb0 100644
--- a/tests/fixed/type-hints.php
+++ b/tests/fixed/type-hints.php
@@ -10,7 +10,7 @@ use Traversable;
 class TraversableTypeHints
 {
     /** @var Traversable */
-    private $parameter;
+    private Traversable $parameter;
 
     /**
      * @param Iterator $iterator
