diff --git a/src/Unleashed/Helpers/SniffLocalCache.php b/src/Unleashed/Helpers/SniffLocalCache.php
index ad0792f..a8eaab3 100644
--- a/src/Unleashed/Helpers/SniffLocalCache.php
+++ b/src/Unleashed/Helpers/SniffLocalCache.php
@@ -14,7 +14,7 @@ use PHP_CodeSniffer\Files\File;
 final class SniffLocalCache
 {
     /** @var array<int, array<string, mixed>> */
-    private static $cache = [];
+    private static array $cache = [];
 
     /**
      * @return mixed
diff --git a/src/Unleashed/Sniffs/Commenting/ForbiddenSingleLineCommentsSniff.php b/src/Unleashed/Sniffs/Commenting/ForbiddenSingleLineCommentsSniff.php
index 3bed058..94d3a48 100644
--- a/src/Unleashed/Sniffs/Commenting/ForbiddenSingleLineCommentsSniff.php
+++ b/src/Unleashed/Sniffs/Commenting/ForbiddenSingleLineCommentsSniff.php
@@ -13,7 +13,7 @@ final class ForbiddenSingleLineCommentsSniff implements Sniff
     public const CODE_COMMENT_FORBIDDEN = 'CommentForbidden';
 
     /** @var string[] */
-    public $forbiddenCommentPatterns = [];
+    public array $forbiddenCommentPatterns = [];
 
     /**
      * @return array<int, (int|string)>
diff --git a/src/Unleashed/Sniffs/Commenting/InheritDocFormatSniff.php b/src/Unleashed/Sniffs/Commenting/InheritDocFormatSniff.php
index 9f7d180..f77cf2e 100644
--- a/src/Unleashed/Sniffs/Commenting/InheritDocFormatSniff.php
+++ b/src/Unleashed/Sniffs/Commenting/InheritDocFormatSniff.php
@@ -13,10 +13,8 @@ final class InheritDocFormatSniff implements Sniff
 
     /**
      * The required style
-     *
-     * @var string
      */
-    public $style = '{@inheritDoc}';
+    public string $style = '{@inheritDoc}';
 
     /**
      * @return array<int, (int|string)>
diff --git a/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php b/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php
index ef06dc6..3106f19 100644
--- a/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php
+++ b/src/Unleashed/Sniffs/Namespaces/FullyQualifiedGlobalFunctionsSniff.php
@@ -11,11 +11,10 @@ use Unleashed\Helpers\UseStatements;
 
 final class FullyQualifiedGlobalFunctionsSniff implements Sniff
 {
-    /** @var bool */
-    public $onlyOptimizedFunctions = false;
+    public bool $onlyOptimizedFunctions = false;
 
     /** @var array<string, bool> */
-    private $optimizedFunctions = [
+    private array $optimizedFunctions = [
         // @see https://github.com/php/php-src/blob/PHP-7.4/Zend/zend_compile.c "zend_try_compile_special_func"
         'array_key_exists'     => true,
         'array_slice'          => true,
diff --git a/src/Unleashed/Sniffs/PHP/ForbiddenClassesSniff.php b/src/Unleashed/Sniffs/PHP/ForbiddenClassesSniff.php
index 674faad..943722a 100644
--- a/src/Unleashed/Sniffs/PHP/ForbiddenClassesSniff.php
+++ b/src/Unleashed/Sniffs/PHP/ForbiddenClassesSniff.php
@@ -19,7 +19,7 @@ final class ForbiddenClassesSniff implements Sniff
      *
      * @var string[]
      */
-    public $forbiddenClasses = [
+    public array $forbiddenClasses = [
         // phpcs:disable Unleashed.PHP.ForbiddenClasses.Forbidden
         \DateTime::class,
         // phpcs:enable
@@ -27,10 +27,8 @@ final class ForbiddenClassesSniff implements Sniff
 
     /**
      * If true, an error will be thrown; otherwise a warning
-     *
-     * @var bool
      */
-    public $error = true;
+    public bool $error = true;
 
     /**
      * Returns an array of tokens this test wants to listen for.
diff --git a/tests/expected_report.txt b/tests/expected_report.txt
index 5c9c68f..a17bed3 100644
--- a/tests/expected_report.txt
+++ b/tests/expected_report.txt
@@ -11,11 +11,11 @@ tests/input/concatenation_spacing.php                 49      0
 tests/input/constants-no-lsb.php                      2       0
 tests/input/constants-var.php                         4       0
 tests/input/ControlStructures.php                     17      2
-tests/input/doc-comment-spacing.php                   16      0
+tests/input/doc-comment-spacing.php                   17      0
 tests/input/doctrine-migration.php                    18      6
 tests/input/duplicate-assignment-variable.php         1       0
 tests/input/EarlyReturn.php                           6       0
-tests/input/example-class.php                         38      0
+tests/input/example-class.php                         41      0
 tests/input/forbidden-comments.php                    14      0
 tests/input/forbidden-functions.php                   13      0
 tests/input/ForbiddenClasses.php                      7       0
@@ -42,16 +42,16 @@ tests/input/strict-functions.php                      4       0
 tests/input/test-case.php                             7       0
 tests/input/trailing_comma_on_array.php               1       0
 tests/input/traits-uses.php                           12      0
-tests/input/type-hints.php                            5       0
+tests/input/type-hints.php                            6       0
 tests/input/UnusedVariables.php                       2       0
 tests/input/use-function.php                          2       0
 tests/input/use-ordering.php                          9       0
 tests/input/useless-semicolon.php                     2       0
 tests/input/UselessConditions.php                     23      0
 ----------------------------------------------------------------------
-A TOTAL OF 429 ERRORS AND 9 WARNINGS WERE FOUND IN 45 FILES
+A TOTAL OF 434 ERRORS AND 9 WARNINGS WERE FOUND IN 45 FILES
 ----------------------------------------------------------------------
-PHPCBF CAN FIX 337 OF THESE SNIFF VIOLATIONS AUTOMATICALLY
+PHPCBF CAN FIX 342 OF THESE SNIFF VIOLATIONS AUTOMATICALLY
 ----------------------------------------------------------------------
 
 
diff --git a/tests/fixed/doc-comment-spacing.php b/tests/fixed/doc-comment-spacing.php
index 2f53db0..728e7f0 100644
--- a/tests/fixed/doc-comment-spacing.php
+++ b/tests/fixed/doc-comment-spacing.php
@@ -25,7 +25,7 @@ class Test
      * @psalm-var list<mixed>
      * @phpstan-var array<mixed>
      */
-    public $foo = [];
+    public array $foo = [];
 
     /**
      * Description
diff --git a/tests/fixed/example-class.php b/tests/fixed/example-class.php
index 0f98ce6..050d89c 100644
--- a/tests/fixed/example-class.php
+++ b/tests/fixed/example-class.php
@@ -18,14 +18,12 @@ class Example implements \IteratorAggregate
 {
     private const VERSION = \PHP_VERSION - (PHP_MINOR_VERSION * 100) - PHP_PATCH_VERSION;
 
-    /** @var int|null */
-    private $foo;
+    private ?int $foo = null;
 
     /** @var string[] */
-    private $bar;
+    private array $bar;
 
-    /** @var bool */
-    private $baz;
+    private bool $baz;
 
     /** @var ControlStructureSniff|int|string|null */
     private $baxBax;
diff --git a/tests/fixed/type-hints.php b/tests/fixed/type-hints.php
index 0e952fc..9824fb0 100644
--- a/tests/fixed/type-hints.php
+++ b/tests/fixed/type-hints.php
@@ -10,7 +10,7 @@ use Traversable;
 class TraversableTypeHints
 {
     /** @var Traversable */
-    private $parameter;
+    private Traversable $parameter;
 
     /**
      * @param Iterator $iterator
